src/ray/gcs/format/gcs.fbs

```diff
@@ -347,3 +347,20 @@ table ActorCheckpointIdData {
   // A list of the timestamps for each of the above `checkpoint_ids`.
   timestamps: [long];
 }
+
+// This enum type is used as object's metadata to indicate the object's creating
+// task has failed because of a certain error.
+// TODO(hchen): We may want to make these errors more specific. E.g., we may want
+// to distinguish between intentional and expected actor failures, and between
+// worker process failure and node failure.
+enum ErrorType:int {
+  // Indicates that a task failed because the worker died unexpectedly while executing it.
+  WORKER_DIED = 1,
+  // Indicates that a task failed because the actor died unexpectedly before finish
+  // executing it.
+  ACTOR_DIED = 2,
+  // Indicates that an object is lost and cannot be reconstructed. Note, this should
+  // only happen to actor objects, because normal task objects can always be
+  // reconstructed.
```

Comment:
Sometimes objects created by `ray.put` on the driver cannot be reconstructed. Does this error apply in that case?

Reply:
Currently, it doesn't. I'll add a TODO here.